app/models/maintenance_tasks/run.rb

```diff
@@ -34,6 +34,8 @@ class Run < ApplicationRecord
       Task.available_tasks.map(&:to_s)
     } }
     validate :csv_attachment_presence, on: :create
+    validate :arguments_match_task_attributes, on: :create,
+      if: -> { arguments.present? }
```

Comment:
We're calling `arguments.keys` in the validation method though - I could return early there, but I thought it was a bit nicer to just use the inline condition here.

Reply:
Yeah personally I'm not a fan of the if option on callbacks when used with a proc. With a symbol I can see why it would be nicer, but overall I don't mind using regular Ruby constructs for this kind of thing.

Also. when checking whether a default value could work, I found that if the task name validation doesn't pass, `arguments_match_task_attributes` fails, we need a rescue similar to what we have in the CSV and task arguments validations.

Finally, shouldn't this logic be included in `validate_task_arguments` instead? That would solve the rescue issue at least, but also it feels kind of the same concern.

Reply:
Yep fair, I'll fold this into `validate_task_arguments` 👍